File: Java/编辑距离/src/com/guoshuman/islandPerimeter.java
package com.guoshuman;

public class islandPerimeter {

    public static void main(String[] args){
        islandPerimeter island = new islandPerimeter();
        int[][] d = new int[][]{{0, 1, 0, 0}, {1, 1, 1, 0}, {0, 1, 0, 0}, {1, 1, 0, 0}};
        int sum = island.perimeter(d);
        System.out.println(sum);
    }

    public static int perimeter(int[][] d){
        int row = d.length;
        int column = d[0].length;
        int repeat = 0;
        int plus = 0;
        int perimeter = 0;
        for(int i=0;i<row;i++){
            for(int j=0;j<column;j++){
                if(d[i][j]==1){
                    plus += 1;
                    if(i != 0 && d[i-1][j]==1){
                        repeat += 1;
                    }
                    if(j!=0 && d[i][j-1]==1){
                        repeat+=1;
                    }
                }
            }
        }
        perimeter = 4*plus-2*repeat;
        return perimeter;
    }
}

File: Python/islandPerimeter.py
 # -*- coding:utf-8 -*-

def islandPerimeter(d):
    row = len(d)
    column = len(d[0])
    perimeter = 0
    plus = 0
    repeat = 0
    for i in xrange(row):
        for j in xrange(column):
            if(d[i][j]==1):
                plus += 1
                if(i!=0 and d[i-1][j]==1):
                    repeat += 1
                if (j!=0 and d[i][j-1]==1):
                    repeat += 1

    perimeter = plus*4-2*repeat
    for i in xrange(row):
        print d[i]
    return perimeter

if __name__ == '__main__':
    d=[[0,1,0,0],[1,1,1,0],[0,1,0,0],[1,1,0,0]]
    sum = islandPerimeter(d)
    print sum
